### PR TITLE
FUSETOOLS-3312 - Fix 'FuseOnOpenShiftTest'

### PR DESCRIPTION
**Pull request by tsedmik**
Signed-off-by: Tomáš Sedmík <tsedmik@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/requirement/OpenShiftRequirement.java [hunk 1]
```diff
@@ -81,8 +81,7 @@ public void fulfill() {
 		wizard.setTextUsername(config.getUsername());
 		wizard.setTextPassword(config.getPassword());
 		wizard.clickAdvancedBTN();
-		wizard.toggleOverrideOcLocationCHB(true);
-		wizard.setOCLocation(config.getOc());
+		wizard.toggleOverrideOcLocationCHB(false);
```
Comment (tsedmik) [inline uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/requirement/OpenShiftRequirement.java]: we leverage `oc` binary on the PATH

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FuseOnOpenShiftTest.java [hunk 2]
```diff
@@ -144,7 +144,9 @@ public void testDeploymentToOpenShift() {
 		// check the deployment
 		explorer.activate();
 		explorer.openPodLog(OPENSHIFT_CONNECTION, OPENSHIFT_PROJECT_NAME, OPENSHIFT_DEPLOYMENT);
-		new WaitUntil(new ConsoleHasText(DEPLOYMENT_OK), TimePeriod.VERY_LONG);
+		new WaitUntil(new ConsoleHasText(DEPLOYMENT_OK), TimePeriod.VERY_LONG, false);
+		explorer.activate();
+		explorer.openPodLog(OPENSHIFT_CONNECTION, OPENSHIFT_PROJECT_NAME, OPENSHIFT_DEPLOYMENT);
```
Comment (tsedmik) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/FuseOnOpenShiftTest.java]: sometimes the console has different output - not the actual pod log, but status of the pod. This is my workaround :-)

**Comment by apupier** [comment]
rebase required due to Target Platform update, see https://github.com/jbosstools/jbosstools-fuse/pull/1597

**Comment by djelinek** [comment]
testPR

**Comment by djelinek** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by djelinek** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
testPR

**Comment by tsedmik** [comment]
Hi @apupier, @djelinek, I opt for merge this PR even though one PR check failed. I do not change anything related to templates (templates job failed). Even _fuse PR_ check is highly unstable and I'm happy to have this one green. Do you agree with me?

**Comment by djelinek** [comment]
Due to instability I vote for yes and agree.

cc @tsedmik @apupier

**Comment by apupier** [comment]
> I opt for merge this PR even though one PR check failed. I do not change anything related to templates (templates job failed). Even _fuse PR_ check is highly unstable and I'm happy to have this one green. Do you agree with me?

:'-( :'-(
ok
the failing tests are different each time so they all passed at least once.

**Pull request merged**